### PR TITLE
Use ESGetToken in JetTagComputerESProducer<T>

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR makes `JetTagComputerESProducer<T>` to use `ESGetToken`s for ESProduct retrieval (as part of #26748). This is done by requiring the `T` to have a nested type `Tokens` that then does the `consumes` call and holds the `ESGetToken`s and is then passed to the constructor of `T` (in case `T` does not consume any ESProducts, it can define `Tokens` as `void` and have no further effect).

In principle I think a "maker" pattern would have been more applicable (`T` would be a "maker" that would be used to produce the concrete `JetTagComputer` in `JetTagComputerESProducer<T>::produce()`), but that would have been much heavier refactoring to which I did not want to go.

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
@makortel please have a look at the output of the header checks:

```
src/RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h:18:32: error: 'GBRWrapperRcd' was not declared in this scope
     edm::ESGetToken<GBRForest, GBRWrapperRcd> gbrForest_;
                                ^~~~~~~~~~~~~
src/RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h:18:45: error: template argument 2 is invalid
     edm::ESGetToken<GBRForest, GBRWrapperRcd> gbrForest_;
                                             ^
```

```
src/RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h:18:32: error: 'GBRWrapperRcd' was not declared in this scope
     edm::ESGetToken<GBRForest, GBRWrapperRcd> gbrForest_;
                                ^~~~~~~~~~~~~
src/RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h:18:45: error: template argument 2 is invalid
     edm::ESGetToken<GBRForest, GBRWrapperRcd> gbrForest_;
                                             ^
```

**Comment by makortel** [comment]
@perrotta Done, thanks

**Comment by makortel** [comment]
@cmsbuild, please test

File: RecoBTau/JetTagComputer/interface/JetTagComputerESProducer.h [hunk 1]
```diff
@@ -2,37 +2,80 @@
...
+ * Those that need the ESGetTokens should define the nested type as a
+ * struct/class containing the ESGetTokens and a constructor taking
+ * edm::ParameterSet and edm::ESConsumesCollector as arguments. In
+ * this case the constructor of ConcreteJetTagComputer takes this
+ * 'Tokens' object as an additional argument.
+ *
+ * Those that do not need ESGetTokens should define the nested type as
+ * void, and in this case no further modifications are needed.
+ */
+namespace jetTagComputerESProducerImpl {
```
Comment (perrotta) [inline RecoBTau/JetTagComputer/interface/JetTagComputerESProducer.h]: As for CMS code rules, please use lowercase letters for namespace's

File: RecoBTau/JetTagComputer/interface/JetTagComputerESProducer.h [hunk 2]
```diff
@@ -2,37 +2,80 @@
...
-    jetTagComputer->initialize(record);
-    jetTagComputer->setupDone();
-    return jetTagComputer;
-  }
-
-private:
-  edm::ParameterSet m_pset;
-};
+/**
+ * The idea here is to provide to implementations for
```
Comment (perrotta) [inline RecoBTau/JetTagComputer/interface/JetTagComputerESProducer.h]: "to implementations" -> "two implementations"

**Comment by makortel** [comment]
@perrotta done

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Kind if technical: it implements what declared in the PR description
- Jenkins test pass and show no differences

**Comment by fabiocos** [comment]
+1

**Pull request merged**